<?xml version="1.0" encoding="utf-8"?>

<!-- <robot xmlns:xacro="http://www.ros.org/wiki/xacro"> -->
<robot xmlns:xacro="https://docs.ros.org/en/eloquent/index.html">

    <xacro:macro name="svaya_transmission" params="hardware_interface">

		<transmission name="trans1">
			<robotNamespace>/${robot_name}</robotNamespace>
			<type>transmission_interface/SimpleTransmission</type>
			<joint name="joint1">
				<hardwareInterface>hardware_interface/${hardware_interface}</hardwareInterface>
			</joint>
			<actuator name="motor1">
				<hardwareInterface>hardware_interface/${hardware_interface}</hardwareInterface>
                <mechanicalReduction>100</mechanicalReduction>
			</actuator>
		</transmission>

		<transmission name="trans2">
			<robotNamespace>/${robot_name}</robotNamespace>
			<type>transmission_interface/SimpleTransmission</type>
			<joint name="joint2">
				<hardwareInterface>hardware_interface/${hardware_interface}</hardwareInterface>
			</joint>
			<actuator name="motor2">
				<hardwareInterface>hardware_interface/${hardware_interface}</hardwareInterface>
                <mechanicalReduction>100</mechanicalReduction>
			</actuator>
		</transmission>

		<transmission name="trans3">
			<robotNamespace>/${robot_name}</robotNamespace>
			<type>transmission_interface/SimpleTransmission</type>
			<joint name="joint3">
				<hardwareInterface>hardware_interface/${hardware_interface}</hardwareInterface>
			</joint>
			<actuator name="motor3">
				<hardwareInterface>hardware_interface/${hardware_interface}</hardwareInterface>
                <mechanicalReduction>101</mechanicalReduction>
			</actuator>
		</transmission>

		<transmission name="trans4">
			<robotNamespace>/${robot_name}</robotNamespace>
			<type>transmission_interface/SimpleTransmission</type>
			<joint name="joint4">
				<hardwareInterface>hardware_interface/${hardware_interface}</hardwareInterface>
			</joint>
			<actuator name="motor4">
				<hardwareInterface>hardware_interface/${hardware_interface}</hardwareInterface>
                <mechanicalReduction>101</mechanicalReduction>
			</actuator>
		</transmission>

		<transmission name="trans5">
			<robotNamespace>/${robot_name}</robotNamespace>
			<type>transmission_interface/SimpleTransmission</type>
			<joint name="joint5">
				<hardwareInterface>hardware_interface/${hardware_interface}</hardwareInterface>
			</joint>
			<actuator name="motor5">
				<hardwareInterface>hardware_interface/${hardware_interface}</hardwareInterface>
                <mechanicalReduction>101</mechanicalReduction>
			</actuator>
		</transmission>

		<transmission name="trans6">
			<robotNamespace>/${robot_name}</robotNamespace>
			<type>transmission_interface/SimpleTransmission</type>
			<joint name="joint6">
				<hardwareInterface>hardware_interface/${hardware_interface}</hardwareInterface>
			</joint>
			<actuator name="motor6">
				<hardwareInterface>hardware_interface/${hardware_interface}</hardwareInterface>
                <mechanicalReduction>101</mechanicalReduction>
			</actuator>
		</transmission>

	</xacro:macro>

</robot>
